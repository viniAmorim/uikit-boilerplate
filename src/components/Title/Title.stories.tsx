import React from 'react';

import { Story, Meta } from '@storybook/react';

import Title, { TitleProps } from './Title';

export default {
  title: 'Atoms/Typography/Title',
  component: Title,
} as Meta;

const Template: Story<TitleProps> = (args) => <Title {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: 'Ad astra per aspera',
};
