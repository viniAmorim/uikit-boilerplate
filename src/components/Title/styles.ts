import { Title as TitleComponent } from '@mantine/core';
import styled from 'styled-components';

export const Title = styled(TitleComponent)`
  color: ${({ theme }) => theme.colors.purple900};
  text-align: center;
  font-family: 'Montserrat';
  font-weight: 700;
  line-height: 42px;
`;
