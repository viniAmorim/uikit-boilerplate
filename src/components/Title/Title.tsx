import React, { BaseHTMLAttributes, ReactNode } from 'react';

import * as Styled from './styles';

export interface TitleProps extends BaseHTMLAttributes<HTMLTitleElement> {
  children: ReactNode;
}

const Title = ({ children }: TitleProps) => {
  return <Styled.Title>{children}</Styled.Title>;
};

export default Title;
