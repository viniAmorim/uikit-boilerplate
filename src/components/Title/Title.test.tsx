import React from 'react';

import { render } from '~/test/utils';

import Title from './Title';

describe('Title', () => {
  it('should render correctly and with the correct styles', () => {
    const { container } = render(<Title>Title</Title>);

    const title = container.firstChild;

    expect(title).toHaveTextContent(/^Title$/);
    expect(title).toBeInTheDocument();
    expect(title).toMatchSnapshot();
    expect(title).toHaveStyle(`
      color: #33245C;
      text-align: center;
      font-weight: 700;
      line-height: 1.3;
  `);
  });
});
