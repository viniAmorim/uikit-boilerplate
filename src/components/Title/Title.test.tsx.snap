// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should render correctly and with the correct styles 1`] = `
<h1
  class="mantine-Text-root mantine-Title-root sc-beqWaB hHMQGN mantine-iuvzso"
>
  Title
</h1>
`;
